### PR TITLE
Remove rasa chat

### PR DESCRIPTION
**Pull request by m-vdb**
Relates to https://github.com/RasaHQ/rasa-plus/pull/846

**Proposed changes**:
- Create a new channel in rasa-plus that extends socketio (e.g. SocketioPro or something like that); this channel is internal for now, no need to add it to built-in channels or to document it. It only extends the blueprint function to add the /chat endpoint – copy the code from OSS there;
- Move the chat CLI command from OSS repo to rasa-plus repo; update the implementation of the refine_cli hook;
- ~Move the /flows endpoint away from the [server.py](https://github.com/RasaHQ/rasa/blob/dm2/rasa/server.py#L1375-L1381); we will probably need to create a hook for this to work;~
- Update pyproject.toml on OSS to remove [this line](https://github.com/RasaHQ/rasa/pull/8897/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R21) and on rasa-plus to add it; it makes sure that chat.html is packaged when released;

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/channels/socketio.py [hunk 1]
```diff
@@ -13,8 +13,6 @@
 
 logger = logging.getLogger(__name__)
 
-CHAT_TEMPLATE_PATH = "/chat.html"
-
 
 class SocketBlueprint(Blueprint):
```
Comment (tmbo) [inline rasa/core/channels/socketio.py:17]: `session_persistance` needs to be set to default to `False` again

Comment (tmbo) [reply rasa/core/channels/socketio.py:17]: Bildschirmfoto 2023-09-21 um 09 19 00

**Review by tmbo — APPROVED** [review]
lgtm :+1:

**Comment by m-vdb** [comment]
@tmbo it looks like we both overlooked the obvious... f9f54bb94f9

**Comment by tmbo** [comment]
lol 😂

**Pull request merged**